lib/tasks/whitehall_importer.rb

```diff
@@ -120,37 +127,93 @@ def create_edition(document, translation, whitehall_edition, edition_number)
         created_at: whitehall_edition["created_at"],
       )
 
-      Edition.create!(
+      edition = Edition.create!(
         document: document,
         number: edition_number,
         revision_synced: true,
         revision: revision,
-        status: Status.new(
-          state: state(whitehall_edition),
-          revision_at_creation: revision,
-        ),
+        status: initial_status(whitehall_edition, revision),
         current: whitehall_edition["id"] == most_recent_edition["id"],
         live: live?(whitehall_edition),
         created_at: whitehall_edition["created_at"],
         updated_at: whitehall_edition["updated_at"],
         created_by_id: user_ids[first_author["whodunnit"]],
         last_edited_by_id: user_ids[last_author["whodunnit"]],
       )
+
+      set_current_status(whitehall_edition, edition) if whitehall_edition["state"] == "withdrawn"
+    end
+
+    def initial_status(whitehall_edition, revision)
+      author = if whitehall_edition["state"] == "withdrawn"
+                 published_author(whitehall_edition)
+               else
+                 current_author(whitehall_edition)
+               end
+
+      raise AbortImportError unless author
+
+      Status.new(
+        state: state(whitehall_edition),
+        revision_at_creation: revision,
+        created_by_id: user_ids[author["whodunnit"]],
+        created_at: author["created_at"],
+      )
+    end
+
+    def set_current_status(whitehall_edition, edition)
+      author = current_author(whitehall_edition)
+
+      raise AbortImportError unless valid_state?(whitehall_edition)
+
+      edition.status = Status.new(
+        state: whitehall_edition["state"],
+        revision_at_creation: edition.revision,
+        created_by_id: user_ids[author["whodunnit"]],
+        created_at: author["created_at"],
+        details: state_details(whitehall_edition, edition),
+      )
+
+      edition.save!
+    end
+
+    def current_author(whitehall_edition)
+      whitehall_edition["revision_history"].select { |h| h["state"] == whitehall_edition["state"] }.last
+    end
+
+    def published_author(whitehall_edition)
+      whitehall_edition["revision_history"].select { |h| h["state"] == "published" }.last
+    end
+
+    def valid_state?(whitehall_edition)
+      return false if whitehall_edition["state"] == "withdrawn" && !whitehall_edition["unpublishing"]
+
+      true
+    end
```

Comment:
I find this method a bit strange as it's entirely withdrawn centric rather than actually covering whether a state is valid. Though I think if you go with my suggestion of `set_withdrawn_status` method you could move that into that and not need this specific method.